cleanup: Support matrix update and code cleanup

karthik-us opened the pull request:
- Updating the support matrix to point to the suppotred ceph release.
- Cleanup the cephfs resize code.

Fixes: https://github.com/ceph/ceph-csi/issues/4168

karthik-us commented:
@Madhu-1 @Rakshith-R PTAL.

karthik-us commented:
*Note to self*: Link the upstream issue in the next update.

Madhu-1 commented on README.md:
```diff
@@ -81,39 +81,39 @@ for its support details.
-|        | Creating and deleting snapshot                            | Alpha          | >= v3.7.0          | >= v1.1.0        | Pacific (>=16.2.0)   | >= v1.17.0         |
-|        | Provision volume from snapshot                            | Alpha          | >= v3.7.0          | >= v1.1.0        | Pacific (>=16.2.0)   | >= v1.17.0         |
-|        | Provision volume from another volume                      | Alpha          | >= v3.7.0          | >= v1.1.0        | Pacific (>=16.2.0)   | >= v1.16.0         |
+| RBD    | Dynamically provision, de-provision Block mode RWO volume | GA             | >= v1.0.0          | >= v1.0.0        | Pacific (>=v16.0.0)  | >= v1.14.0         |
```
`Feature Status` need to be updated to change the status to next version based on the support we have (This is for followup PR)

Madhu-1 commented on internal/cephfs/core/volume.go:
```diff
@@ -199,16 +199,13 @@ func (s *subVolumeClient) GetSubVolumeInfo(ctx context.Context) (*Subvolume, err
 	// unsupported as per the state of the cluster.
-	resizeState                 operationState
-	subVolMetadataState         operationState
 	subVolSnapshotMetadataState operationState
```
please remove `subVolSnapshotMetadataState` as well?

can you also check do we have metadata support in `v16.0.0` as well?

karthik-us replied:
metadata & snapshot metadata support are added on pacific from v16.2.11 and on quincy it is from v17.2.1. Should we keep both these checks for some more time in that case, if someone uses versions lower than that?

Madhu-1 replied:
If that is a supported version in ceph we need to keep it as well. ie (< 16.2.11) and (<17.2.1)

karthik-us replied:
As per the docs 16.2.* and 17.2.* are still supported. So we need to keep these checks for some more time.
Will update support matrix as well to point to >=v16.2.0 for all the pacific versions.

Madhu-1 replied:
i believe in `ResizeSubVolume`  we can remove the check and also remove the call for createVolume. can you please check on that

karthik-us replied:
Yes, the support for it is present in all the active releases. Removed it now.

Madhu-1 commented on internal/cephfs/core/volume.go:
```diff
@@ -199,16 +199,13 @@ func (s *subVolumeClient) GetSubVolumeInfo(ctx context.Context) (*Subvolume, err
 	unsupported
 )
 
 type localClusterState struct {
```
`localClusterState` might need to be removed as well as we don't require it anymore.

karthik-us replied:
Since the checks are needed, keeping it as it is for the time being.

Madhu-1 commented on internal/cephfs/core/volume.go:
```diff
@@ -268,39 +265,22 @@ func (s *subVolumeClient) ExpandVolume(ctx context.Context, bytesQuota int64) er
-// CreateVolume to resize the subvolume.
+// ResizeVolume will use ceph fs subvolume resize command to resize the subvolume.
 func (s *subVolumeClient) ResizeVolume(ctx context.Context, bytesQuota int64) error {
 	newLocalClusterState(s.clusterID)
```
`newLocalClusterState(s.clusterID)` also need to be removed

karthik-us replied:
Done.

karthik-us committed:
doc: Support matrix update

Updating the support matrix to point to the suppotred ceph release.

Fixes: #4168
Signed-off-by: karthik-us <ksubrahm@redhat.com>

karthik-us committed:
cleanup: cephfs resize code cleanup

The ceph fs subvolume resize support is available
in all the active ceph releases. Hence removing the
code to check the supportability of the feature.

Signed-off-by: karthik-us <ksubrahm@redhat.com>

karthik-us commented:
/test ci/centos/mini-e2e/k8s-1.28

karthik-us commented:
@Rakshith-R @nixpanic @yati1998 PTAL.

iPraveenParihar reviewed (dismissed):
Thanks, LGTM!

Rakshith-R commented:
@karthik-us 
Can you please update PendingRelease.md too ?

karthik-us committed:
doc: update pending release notes

Adding a note on supported ceph releases.

Signed-off-by: karthik-us <ksubrahm@redhat.com>

karthik-us commented:
> @karthik-us Can you please update PendingRelease.md too ?

@Rakshith-R just the mention of removal of resize code with the reason suffice in the pending release notes as the desscription for the PR right? under the **Breaking Chnages**.

Rakshith-R commented:
> > @karthik-us Can you please update PendingRelease.md too ?
> 
> @Rakshith-R just the mention of removal of resize code with the reason suffice in the pending release notes as the desscription for the PR right? under the **Breaking Chnages**.

The code change is internal only.
No, I meant regarding the support status change  for ceph version.

karthik-us commented:
> > > @karthik-us Can you please update PendingRelease.md too ?
> > 
> > 
> > @Rakshith-R just the mention of removal of resize code with the reason suffice in the pending release notes as the desscription for the PR right? under the **Breaking Chnages**.
> 
> The code change is internal only. No, I meant regarding the support status change for ceph version.

Done.

nixpanic commented:
@mergifyio rebase

nixpanic commented:
@mergifyio queue
